### PR TITLE
Photogram industrial authorization bl

### PR DESCRIPTION
**Pull request by tluu5**

### CODE REVIEW

**Comment by armstrca** [comment]
You're supposed to have one branch for non-Pundit authorization, reviewed by @marywebby, and one branch for Pundit authorization, reviewed by me. It seems like this is maybe a combination of both?

**Comment by armstrca** [comment]
Either way, Alice seems to be able to edit and delete anyone's posts and comments still:
Screenshot 2024-11-01 113410

**Review by armstrca — APPROVED** [review]
Sick

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,10 +3,26 @@ class ApplicationController < ActionController::Base
...
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:username, :private, :name, :bio, :website, :avatar_image])
     devise_parameter_sanitizer.permit(:account_update, keys: [:username, :private, :name, :bio, :website, :avatar_image])
   end
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+    def user_not_authorized
```
Comment (jessjno) [inline app/controllers/application_controller.rb:23]: nice job creating the private method to use an after_action

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -50,6 +53,7 @@ def update
 
   # DELETE /comments/1 or /comments/1.json
   def destroy
+    authorize(@comment)
```
Comment (jessjno) [inline app/controllers/comments_controller.rb:56]: you can also create a private method with a before_action for authorize(@comment) so you're not repeating this so much

File: app/controllers/follow_requests_controller.rb
Comment (jessjno) [inline app/controllers/follow_requests_controller.rb:1]: don't forget to remove commented out code if you no longer need it

File: app/models/comment.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: "User", counter_cache: true
   belongs_to :photo, counter_cache: true
+  has_one :owner, through: :photo
```
Comment (jessjno) [inline app/models/comment.rb:25]: good job setting up the association

File: app/policies/application_policy.rb
Comment (jessjno) [inline app/policies/application_policy.rb:1]: nice formatting it looks clean

File: app/policies/user_policy.rb
Comment (jessjno) [inline app/policies/user_policy.rb:1]: looks good

File: config/routes.rb
Comment (jessjno) [inline config/routes.rb:1]: good job setting up the resources

**Review by jessjno — APPROVED** [review]
looks good!

**Pull request merged**